const statusCode = require("../../core/status/statusCode");
const customerReceiptServc = require("../../core/services/customerReceiptServices/customerReceiptServices");

const createErrorMessage = (message, data) => {
  return {
    status: statusCode,
    data: data,
    message: message,
    error: true,
  };
};

module.exports = {
  async createCustomerReceiptContrlr(req, res) {
    try {
      let response = await customerReceiptServc.createCustomerReceiptsSrvc(
        req.body
      );
      return res.status(response.status).send(response);
    } catch (error) {
      console.error(error);
      const newError = createErrorMessage();
      newError.data = error;
      newError.message = "Create Customer Controller Internal Server Error";
      newError.status = statusCode.internalServerError;
      newError.error = true;
      return res.status(newError.status).json(newError);
    }
  },

  async showCustomerReceiptContrlr(req, res) {
    try {
      let response = await customerReceiptServc.showCustomerReceiptsSrvc();
      return res.status(response.status).send(response);
    } catch (error) {
      console.error(error);
      const newError = createErrorMessage();
      newError.data = error;
      newError.message = "Show Customer Controller Internal Server Error";
      newError.status = statusCode.internalServerError;
      newError.error = true;
      return res.status(newError.status).json(newError);
    }
  },

  async showAllCustomerReceiptContrlr(req, res) {
    try {
      let response = await customerReceiptServc.showAllCustomerReceiptsSrvc();
      return res.status(response.status).send(response);
    } catch (error) {
      console.error(error);
      const newError = createErrorMessage();
      newError.data = error;
      newError.message = "Show All Controller Internal Server Error";
      newError.status = statusCode.internalServerError;
      newError.error = true;
      return res.status(newError.status).json(newError);
    }
  },

  async showSingleCustomerReceiptContrlr(req, res) {
    try {
      let response = await customerReceiptServc.showSingleCustomerReceiptsSrvc(
        req.params
      );
      return res.status(response.status).send(response);
    } catch (error) {
      console.error(error);
      const newError = createErrorMessage();
      newError.data = error;
      newError.message = "Show Single Controller Internal Server Error";
      newError.status = statusCode.internalServerError;
      newError.error = true;
      return res.status(newError.status).json(newError);
    }
  },

  async deleteSingleCustomerReceiptContrlr(req, res) {
    try {
        let response = await customerReceiptServc.deleteCustomerReceiptsSrvc(req.params);
      return res.status(response.status).send(response);
    } catch (error) {
      console.error(error);
      const newError = createErrorMessage();
      newError.data = error;
      newError.message = "Delete Controller Internal Server Error";
      newError.status = statusCode.internalServerError;
      newError.error = true;
      return res.status(newError.status).json(newError);
    }
  },

  async updateSingleCustomerReceiptContrlr(req, res) {
    try {
      let response = await customerReceiptServc.updateCustomerReceiptsSrvc(
        req.body,
        req.params
      );
      return res.status(response.status).send(response);
    } catch (error) {
      console.error(error);
      const newError = createErrorMessage();
      newError.data = error;
      newError.message = "Update Controller Internal Server Error";
      newError.status = statusCode.internalServerError;
      newError.error = true;
      return res.status(newError.status).json(newError);
    }
  },
};
